src/Backups/BackupsWorker.cpp

```diff
@@ -183,6 +173,17 @@ OperationID BackupsWorker::startMakingBackup(const ASTPtr & query, const Context
     String backup_name_for_logging = backup_info.toStringForLogging();
     try
     {
+        /// Check if there are no concurrent backups
+        if (num_active_backups && !allow_concurrent_backups)
+        {
+            /// If its an internal backup and we currently have 1 active backup, it could be the original query, validate using backup_uuid
+            if (!(num_active_backups == 1 && backup_settings.internal && getAllActiveBackupInfos().at(0).id == toString(*backup_settings.backup_uuid)))
+            {
+                addInfo(backup_id, backup_name_for_logging, backup_settings.internal, BackupStatus::BACKUP_FAILED);
+                throw Exception(ErrorCodes::CONCURRENT_ACCESS_NOT_SUPPORTED, "Concurrent backups not supported, turn on setting 'allow_concurrent_backups'");
```

Comment:
I think a better place for those checks would be after the call

```
addInfo(backup_id, backup_name_for_logging, backup_settings.internal, BackupStatus::CREATING_BACKUP);
```

below. Because if you move your check there all you will have to do is to throw an exception without this extra `addInfo()` call.